src/vs/workbench/services/editor/browser/codeEditorService.ts

```diff
@@ -55,7 +60,16 @@ export class CodeEditorService extends CodeEditorServiceImpl {
 			const textOptions = TextEditorOptions.create(input.options);
 			textOptions.apply(targetEditor, ScrollType.Smooth);
 
-			return Promise.resolve(targetEditor);
+			return targetEditor;
+		}
+
+		// CompositeCodeEditor
+		const control = this.editorService.activeEditorPane?.getControl();
+		if (ICompositeCodeEditor.is(control)) {
+			const editor = control.activate(input);
+			if (isCodeEditor(editor)) {
+				return editor;
+			}
```

Comment:
fyi @bpasero - this the current usage of `ICompositeCodeEditor` (defined in src/vs/workbench/common/editor.ts). I have changed back (6cd184e2c4f1be9e01567174dfbf6fd729c64ee5) the workbench editor service to not know anything about the composite editor (despite it knowing about code editor and diff editor) and kept it local to the code editor service.

Reply:
@jrieken yeah makes sense. I think it is up to you to keep it like this or add methods for convenience to the editor service. I only exposed them from there due to the number of users, but if the composite code editor usage is little, we can leave it like this.